common/app/common/i18n/CompositeI18nResolver.java

```diff
@@ -0,0 +1,34 @@
+package common.i18n;
+
+import io.sphere.sdk.models.Base;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ * Resolves the i18n messages by trying to resolve them with each given i18n resolver until one of them succeeds.
+ * This class allows to override i18n resolvers.
+ */
+public final class CompositeI18nResolver extends Base implements I18nResolver {
+    private final List<I18nResolver> i18nResolvers;
+
+    private CompositeI18nResolver(final List<I18nResolver> i18nResolvers) {
+        this.i18nResolvers = i18nResolvers;
+    }
+
+    @Override
+    public Optional<String> get(final String bundle, final String key, final Locale locale, final Object... args) {
```

Comment:
Yes. I put args at the last moment so I just could put it in the end because of the ellipsis. Also I was still rethinking about making a class that already contains the locale... Well, I'll start changing the order.